pkg/runner/runner.go

```diff
@@ -19,13 +19,13 @@ import (
 func StartYaraHunter(opts *config.Options, config *config.Config, newwg *sync.WaitGroup) {
 	defer newwg.Done()
 
-	if *opts.SocketPath != "" {
-		err := server.RunGrpcServer(opts, config, constants.PLUGIN_NAME)
-		if err != nil {
-			log.Fatal("main: failed to serve: %v", err)
-		}
-	} else {
+	if *opts.SocketPath == "" {
 		runOnce(opts, config)
+		return
+	}
+
+	if err := server.RunGrpcServer(opts, config, constants.PluginName); err != nil {
+		log.Errorf("main: failed to serve: %v", err)
```

Comment:
We should still bail off if the gRPC server is not able to spin up.
Either we `Fatal` here, or we propagate the error up to main (preferred approach)